src/golang/cmd/server/handler/v2/node_artifact_result_content_get.go

```diff
@@ -0,0 +1,214 @@
+package v2
+
+import (
+	"context"
+	"mime/multipart"
+	"net/http"
+
+	"github.com/aqueducthq/aqueduct/cmd/server/handler"
+	"github.com/aqueducthq/aqueduct/cmd/server/request/parser"
+	aq_context "github.com/aqueducthq/aqueduct/lib/context"
+	"github.com/aqueducthq/aqueduct/lib/database"
+	"github.com/aqueducthq/aqueduct/lib/errors"
+	"github.com/aqueducthq/aqueduct/lib/models/shared"
+	"github.com/aqueducthq/aqueduct/lib/repos"
+	"github.com/aqueducthq/aqueduct/lib/storage"
+	"github.com/aqueducthq/aqueduct/lib/workflow/artifact"
+	"github.com/google/uuid"
+)
+
+const (
+	formIsDownsampledField = "is_downsampled"
+	formContentField       = "content"
+)
+
+// This file should map directly to
+// src/ui/common/src/handlers/NodeArtifactResultContentGet.tsx
+//
+// Route: /api/v2/workflow/{workflowID}/dag/{dagID}/node/artifact/{nodeID}/result/{nodeResultID}/content
+// Method: GET
+// Params:
+//	`workflowID`: ID for `workflow` object
+//  `dagID`: ID for `workflow_dag` object
+//	`nodeID`: ID for node object
+//	`nodeResultID`: ID for node result object
+// Request:
+//	Headers:
+//		`api-key`: user's API Key
+// Response:
+//	Body:
+//		`response.Content`
+
+type NodeArtifactResultContentGetHandler struct {
+	nodeGetHandler
+	handler.GetHandler
+
+	Database database.Database
+
+	WorkflowRepo       repos.Workflow
+	DAGRepo            repos.DAG
+	ArtifactRepo       repos.Artifact
+	ArtifactResultRepo repos.ArtifactResult
+}
+
+type nodeResultGetArgs struct {
+	*aq_context.AqContext
+	workflowID   uuid.UUID
+	dagID        uuid.UUID
+	nodeID       uuid.UUID
+	nodeResultID uuid.UUID
+}
+
+type nodeResultGetResponse struct {
+	IsDownsampled bool   `json:"is_downsampled"`
+	Content       []byte `json:"content"`
+}
+
+func (*NodeArtifactResultContentGetHandler) Name() string {
+	return "NodeArtifactResultContentGet"
+}
+
+func (h *NodeArtifactResultContentGetHandler) Prepare(r *http.Request) (interface{}, int, error) {
+	aqContext, statusCode, err := aq_context.ParseAqContext(r.Context())
+	if err != nil {
+		return nil, statusCode, err
+	}
+
+	workflowID, err := (parser.WorkflowIDParser{}).Parse(r)
+	if err != nil {
+		return nil, http.StatusBadRequest, err
+	}
+
+	dagID, err := (parser.DagIDParser{}).Parse(r)
+	if err != nil {
+		return nil, http.StatusBadRequest, err
+	}
+
+	nodeID, err := (parser.NodeIDParser{}).Parse(r)
+	if err != nil {
+		return nil, http.StatusBadRequest, err
+	}
+
+	nodeResultID, err := (parser.NodeResultIDParser{}).Parse(r)
+	if err != nil {
+		return nil, http.StatusBadRequest, err
+	}
+
+	return &nodeResultGetArgs{
+		AqContext:    aqContext,
+		workflowID:   workflowID,
+		dagID:        dagID,
+		nodeID:       nodeID,
+		nodeResultID: nodeResultID,
+	}, http.StatusOK, nil
+}
+
+func (h *NodeArtifactResultContentGetHandler) Perform(ctx context.Context, interfaceArgs interface{}) (interface{}, int, error) {
+	args := interfaceArgs.(*nodeResultGetArgs)
```

Comment:
Not much need to review here, the implementation is copied from current `get_artifact_result` route. We will add necessary fix once we start integrate with UI.